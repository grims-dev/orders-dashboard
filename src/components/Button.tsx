import styled from "styled-components";

interface BtnProps {
  color?: string;
}

const Button = styled.button<BtnProps>`
  display: block;
  border: 0;
  cursor: pointer;
  transition: all 0.25s ease-out;
  padding: 12px 18px;
  margin: 12px 0px 12px;
  font-size: 14px;
  font-weight: 700;
  color: #fff;
  background-color: ${(props) => (props.color ? props.color : "#3498db")};
  border-radius: 4px;
`;

export default Button;
