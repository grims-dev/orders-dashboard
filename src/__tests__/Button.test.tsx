import { render } from "enzyme";
import * as React from "react";
import toJSON from "enzyme-to-json";
import Button from "../components/Button";

describe("Button renders successfully", () => {
  it("renders without crashing", () => {
    const wrapper = render(<Button />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
