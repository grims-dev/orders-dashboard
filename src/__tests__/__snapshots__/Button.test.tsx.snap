// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders successfully renders without crashing 1`] = `
<button
  class="sc-AxjAm cTkZEu"
/>
`;
